lib/graphql/plug.ex

```diff
@@ -26,18 +26,22 @@ defmodule GraphQL.Plug do
     pass: ["*/*"],
     json_decoder: Poison
 
-  plug GraphQL.Plug.Endpoint
-  # TODO extract
-  # plug GraphQL.Plug.GraphiQL
-
-  # TODO remove duplication call GraphQL.Plug.Helper.extract_init_options/1 here
   def init(opts) do
     schema = case Keyword.get(opts, :schema) do
       {mod, func} -> apply(mod, func, [])
       s -> s
     end
+
     root_value = Keyword.get(opts, :root_value, %{})
```

Comment:
As discussed I think that the core params need to be extracted in the core `Endpoint` plug and this decorates and adds the extra behaviour `graphiql` in this case